packages/react-dom/src/__tests__/ReactDOMFloat-test.js

```diff
@@ -28,3718 +34,3862 @@ let hasErrored = false;
 let fatalError = undefined;
 const renderOptions = {};
 
-describe('ReactDOMFloat', () => {
-  beforeEach(() => {
-    jest.resetModules();
-    JSDOM = require('jsdom').JSDOM;
-    Scheduler = require('scheduler');
-    React = require('react');
-    ReactDOM = require('react-dom');
-    ReactDOMClient = require('react-dom/client');
-    ReactDOMFizzServer = require('react-dom/server');
-    Stream = require('stream');
-    Suspense = React.Suspense;
-
-    textCache = new Map();
-
-    // Test Environment
-    const jsdom = new JSDOM(
-      '<!DOCTYPE html><html><head></head><body><div id="container">',
-      {
-        runScripts: 'dangerously',
-      },
-    );
-    document = jsdom.window.document;
-    container = document.getElementById('container');
-
-    buffer = '';
-    hasErrored = false;
-
-    writable = new Stream.PassThrough();
-    writable.setEncoding('utf8');
-    writable.on('data', chunk => {
-      buffer += chunk;
-    });
-    writable.on('error', error => {
-      hasErrored = true;
-      fatalError = error;
-    });
-  });
+describe(`ReactDOMFloat`, () => {
+  ['browser', 'node'].forEach(runtime => {
+    describe(runtime, () => {
+      beforeEach(() => {
+        jest.resetModules();
+        JSDOM = require('jsdom').JSDOM;
+        Scheduler = require('scheduler');
+        React = require('react');
+        ReactDOM = require('react-dom');
+        ReactDOMClient = require('react-dom/client');
+        ReactDOMFizzServer =
+          runtime === 'node'
+            ? require('react-dom/server')
+            : require('react-dom/server.browser');
```

Comment:
This actually may not work for Jest. We need to come up with another way to fully exercise our tests in a variety of runtimes but this may not be the right approach. Do you have insight into how to resolve the test failures that are occurring?

cc @sebmarkbage would be good to get your insight on what Jimmy should do for tests for this kind of change. I love the ambition here but I know there are a lot of complications when trying to import from more than one runtime in a single test file